library/multichoicerecyclerview/src/main/java/com/davidecirillo/multichoicerecyclerview/MultiChoiceAdapter.java

```diff
@@ -80,8 +83,14 @@ public void selectAll() {
      * @return true if has been selected false otherwise
      */
     public boolean select(int position) {
-        if (mIsInMultiChoiceMode || mIsInSingleClickMode && mItemList.containsKey(position)) {
-            perform(Action.SELECT, position, true, true);
+        if (mItemList.containsKey(position)) {
+            if (mIsInSingleClickMode || mIsInMultiChoiceMode) {
+                perform(Action.SELECT, position, true, true);
+                return true;
+            }
+        } else {
+            mItemList.put(position, State.INACTIVE);
```

Comment:
So, there reason why I added this is because it can happen that the user add the list of item to the adapter later and at the `onAttachToRecyclerView` the `getItemCount` is 0. So because of the way the library works, every bind i'm updating the view with the correct state, and adding this will also add the position in the list if not there. 

For consistency i've added this on the `processClick`, `processUpdate` and `select` but in theory we can just have it in the `processUpdate` because is the one in charge to update the view.

Do you reckon i should only add it to the `processUpdate` one?